source/module_lr/potentials/pot_hxc_lrtd.cpp

```diff
@@ -6,28 +6,69 @@
 #include "module_hamilt_general/module_xc/xc_functional.h"
 #include <set>
 #include "module_lr/utils/lr_util.h"
-
+#include "module_io/cube_io.h"
+#include "module_hamilt_pw/hamilt_pwdft/global.h"    // tmp, for pgrid
 #define FXC_PARA_TYPE const double* const rho, ModuleBase::matrix& v_eff, const std::vector<int>& ispin_op = { 0,0 }
 namespace LR
 {
     // constructor for exchange-correlation kernel
     PotHxcLR::PotHxcLR(const std::string& xc_kernel_in, const ModulePW::PW_Basis* rho_basis_in, const UnitCell* ucell_in,
-        const Charge* chg_gs/*ground state*/, const SpinType& st_in)
-        :xc_kernel(xc_kernel_in), tpiba_(ucell_in->tpiba), spin_type_(st_in)
+        const Charge* chg_gs/*ground state*/, const Parallel_Grid& pgrid,
+        const SpinType& st_in, const std::vector<std::string>& init_fxc)
+        :xc_kernel(xc_kernel_in), tpiba_(ucell_in->tpiba), spin_type_(st_in),
+        xc_kernel_components_(*rho_basis_in)
     {
         this->rho_basis_ = rho_basis_in;
         this->nrxx = chg_gs->nrxx;
         this->nspin = (PARAM.inp.nspin == 1 || (PARAM.inp.nspin == 4 && !PARAM.globalv.domag && !PARAM.globalv.domag_z)) ? 1 : 2;
 
         this->pot_hartree = LR_Util::make_unique<elecstate::PotHartree>(this->rho_basis_);
+
         std::set<std::string> local_xc = { "lda", "pbe", "hse" };
         if (local_xc.find(this->xc_kernel) != local_xc.end())
         {
             XC_Functional::set_xc_type(this->xc_kernel);    // for hse, (1-alpha) and omega are set here
             this->xc_type_ = XCType(XC_Functional::get_func_type());
-#ifdef USE_LIBXC
             this->set_integral_func(this->spin_type_, this->xc_type_);
-            this->xc_kernel_components_.f_xc_libxc(nspin, ucell_in->omega, ucell_in->tpiba, chg_gs);
+
+            if (init_fxc[0] == "file_fxc")
+            {
+                assert(this->xc_kernel == "lda");
+                const int spinsize = (1 == nspin) ? 1 : 3;
+                std::vector<double> v2rho2(spinsize * nrxx);
+                // read fxc adn add to xc_kernel_components
+                assert(init_fxc.size() >= spinsize + 1);
+                for (int is = 0;is < spinsize;++is)
+                {
+                    double ef = 0.0;
+                    int prenspin = 1;
+                    std::vector<double> v2rho2_tmp(nrxx);
+                    ModuleIO::read_vdata_palgrid(pgrid, GlobalV::MY_RANK, GlobalV::ofs_running, init_fxc[is + 1],
```

Comment:
I think it's nothing strange that different processor does different thing to set its data member. But I think it's better to use `GlobalV::MY_RANK` inside `read_vdata_palgrid` rather than passing it as a parameter.
What's your opinion?

Reply:
@maki49 I do not have quite specific preference on this, because there are two possible interpretations:
1. Once you instantiate a class with rank number, it means "explicitly, the instantiation will have different behavior depending on rank number". And following implemention should be consistent with this idea (more or less)
2. Once you do not include the rank number in constructor, it means "an instance of class is created, before further 'heavy' initialization, its properties are independent with serial or parallel run". To avoid use get number from GlobalV directly, still, you can use a function called "init" instead presently, in that function you code the rank-specific behavior and claim "it is the only part that this class works differently/have different behavior/different workflow".

I do not suggest you use GlobalV inside some basic functions.

Reply:
I see. There's no such an "Init" in `PotHxcLR`, and the constructor depending on the rank number does not block the serial version, where MY_RANK is always zero.

Reply:
@maki49 what I am talking about is the concept "based on what quantities can an instance of this class can construct", this question is another expression of "what is the instance of this class".

Reply:
I see what you mean.
Here the class behavior is not depend on the rank number once initialized, but the way to assign its local data depends on the rank number.Therefore the rank number is only used in the constructor, but not passed into any other members. So I feel a little confused: what's the problem, and how should I change?